esp-rtos/src/esp_radio/mod.rs

```diff
@@ -160,10 +160,14 @@ impl SemaphoreImplementation for Semaphore {
     }
 
     unsafe fn try_give_from_isr(semaphore: SemaphorePtr, _hptw: Option<&mut bool>) -> bool {
+        // FIXME: when the interrupt handler that calls this interrupts the idle hook, the OS
```

Comment:
This prevents us from using CompatQueue but otherwise doesn't cause problems with esp-radio, at least not currently.